alfresco-search/src/main/java/org/apache/solr/handler/component/AlfrescoLukeRequestHandler.java

```diff
@@ -461,19 +461,14 @@ private static String getFieldFlags(SchemaField f) {
 	// to do it this way.
 	private static Document getFirstLiveDoc(Terms terms, LeafReader reader)
 			throws IOException {
-		PostingsEnum postingsEnum = null;
 		TermsEnum termsEnum = terms.iterator();
-		BytesRef text;
 		// Deal with the chance that the first bunch of terms are in deleted
 		// documents. Is there a better way?
-		for (int idx = 0; idx < 1000 && postingsEnum == null; ++idx) {
-			text = termsEnum.next();
-			if (text == null) { // Ran off the end of the terms enum without
-								// finding any live docs with that field in
-								// them.
+		for (int idx = 0; idx < 1000; ++idx) {

```

Comment:
I made a mistake here. I assumed that postingsEnum must be null, but actually after the first iteration, postingsEnum must *not* be null.  I should fix this by removing the for loop.